qemu/tests/balloon_stress.py

```diff
@@ -34,27 +79,28 @@ def run(test, params, env):
 
     driver_name = params["driver_name"]
     utils_test.qemu.setup_win_driver_verifier(driver_name, vm, timeout)
+    run_video(vm)
+    time.sleep(5)
+    repeat_times = int(params.get("repeat_times", 500))
+    time_for_video = float(params.get("time_for_video", 240))
     balloon_test = BallooningTestWin(test, params, env)
-
-    video_play = utils_misc.InterruptedThread(
-        utils_test.run_virt_sub_test, (test, params, env),
-        {"sub_type": params.get("sub_test")})
-    video_play.start()
-    error_context.context("Run video background", logging.info)
-    check_playing_cmd = params["check_playing_cmd"]
-    running = utils_misc.wait_for(
-        lambda: utils_misc.get_guest_cmd_status_output(
-            vm, check_playing_cmd)[0] == 0, first=60, timeout=600)
-    if not running:
-        raise exceptions.TestError("Video is not playing")
+    min_sz, max_sz = balloon_test.get_memory_boundary()
 
     error_context.context("balloon vm memory in loop", logging.info)
-    repeat_times = int(params.get("repeat_times", 10))
-    logging.info("repeat times: %d" % repeat_times)
-    min_sz, max_sz = balloon_test.get_memory_boundary()
-    while repeat_times:
+    start_time = time.time()
+    for i in xrange(repeat_times):
+        logging.info("repeat times: %d" % i)
+        play_duration = time.time() - start_time
+        if i == 0 or (play_duration > time_for_video):
+            no_video = utils_misc.wait_for(lambda: not check_video_status(vm),
+                                           30, 5, 2)
+            if not no_video:
+                process = check_video_status(vm)
+                stop_video(process)
+            run_video(vm)
+            start_time = time.time()
```

Comment:
It doesn't make sense to control the video play workflow here, suggest moving into `run_video`.